src/wscript_build

```diff
@@ -1,59 +1,146 @@
+import platform
+
 all_sources = bld.path.ant_glob('*.cpp scene-ideas/*.cc scene-terrain/*.cpp')
 common_sources = [f for f in all_sources if f.name.find('canvas-') == -1 and
                                             f.name.find('android') == -1 and
                                             f.name.find('native-state-') == -1 and
                                             f.name.find('gl-state-') == -1 and
                                             f.name.find('main.cpp') == -1]
-common_uselibs = ['libpng']
+common_uselibs = ['libpng', 'libjpeg-turbo']
 common_defines = ['USE_EXCEPTIONS']
 
 libmatrix_sources = [f for f in bld.path.ant_glob('libmatrix/*.cc')
                      if not f.name.endswith('test.cc')]
 
 common_flavor_sources = ['main.cpp', 'canvas-generic.cpp']
 
+libpng_sources = [f for f in bld.path.ant_glob('libpng/*.c')]
+zlib_sources = [f for f in bld.path.ant_glob('zlib/*.c')]
+
+# Some sources in libjpeg-turbo are included like headers.
+libjpeg_sources = [
+  'jcapimin',
+  'jcapistd',
+  'jccoefct',
+  'jccolor',
+  'jcdctmgr',
+  'jchuff',
+  'jcinit',
+  'jcmainct',
+  'jcmarker',
+  'jcmaster',
+  'jcomapi',
+  'jcparam',
+  'jcphuff',
+  'jcprepct',
+  'jcsample',
+  'jdapimin',
+  'jdapistd',
+  'jdatadst',
+  'jdatasrc',
+  'jdcoefct',
+  'jdcolor',
+  'jddctmgr',
+  'jdhuff',
+  'jdinput',
+  'jdmainct',
+  'jdmarker',
+  'jdmaster',
+  'jdmerge',
+  'jdphuff',
+  'jdpostct',
+  'jdsample',
+  'jerror',
+  'jfdctflt',
+  'jfdctfst',
+  'jfdctint',
+  'jidctflt',
+  'jidctfst',
+  'jidctint',
+  'jidctred',
+  'jmemmgr',
+  'jmemnobs',
+  'jquant1',
+  'jquant2',
+  'jutils',
+  'jsimd_none',
+]
+
+libjpeg_turbo_sources = ['libjpeg-turbo/%s.c' % f for f in libjpeg_sources]
+
+platform_libs = []
+platform_includes = []
+
+if platform.system() == 'Windows':
```

Comment:
I am hesitant about using "platform" for this, since this makes compilation with, e.g. mingw32, impossible. See my later comments in 'wscript' for a possible alternative.

Reply:
(but as I also mention below, not a blocker)

Reply:
Done.